### PR TITLE
[Neo Nullable ]Neo nullable

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This pr focus on making neo nullable.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Test fail, need check.

**Test Configuration**:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo/Cryptography/ECC/ECPoint.cs [hunk 1]
```diff
@@ -23,9 +23,9 @@ namespace Neo.Cryptography.ECC
     /// </summary>
     public class ECPoint : IComparable<ECPoint>, IEquatable<ECPoint>, ISerializable
     {
-        internal ECFieldElement X, Y;
+        internal ECFieldElement? X, Y;
```
Comment (shargon) [inline src/Neo/Cryptography/ECC/ECPoint.cs:26]: It should not be null

Comment (Jim8y) [reply src/Neo/Cryptography/ECC/ECPoint.cs:26]: check the class, it has null assignment and null check.

Comment (shargon) [reply src/Neo/Cryptography/ECC/ECPoint.cs:26]: Maybe we can fix it, Why we want a point with null X and Y ? xD

Comment (Jim8y) [reply src/Neo/Cryptography/ECC/ECPoint.cs:26]: Infinity

**Comment by Jim8y** [comment]
use https://github.com/neo-project/neo/pull/3125 instead

**Pull request closed without merging**